testutil/dirmap.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 The Authors (see AUTHORS file)
```

Comment:
I'm partial to naming this package "abctest" instead of "testutil". You're already aliasing the package since it conflicts with other "testutil" packages, but I think "abctest" is more descriptive, especially if we plan on having others use it.